### DISCUSSION TITLE
Panorama commit and push

### DISCUSSION THREAD

**Discussion by ruddan** [PaloAltoNetworks/pan-os-python#384] [Q&A]
Hi!

I'm looking for a class to do a CommitAndPush job from Panorama, as in commit changes to Panorama and push changes to affected firewalls.

To achieve this now I need to:

1. Pass `panos.panorama.PanoramaCommit` to `Panorama.commit()` to commit changes to Panorama
2. Pass `panos.panorama.PanoramaCommitAll` to `Panorama.commit()` to push changes to **specified location** (argument to push changes to affected firewalls would be great as well)

This feels quite inefficient since we have an option in Panorama to CommitAndPush.

**Comment by danielyeriksson** [comment]
Any news here?

**Comment by jcrubaugh** [comment]
here is how i did the panorama commit 
#this is the api call for the script to use to do the basic panorama commmit 
```commit_panorama_api = "https://panoramaname/api/?type=commit&cmd=<commit></commit>&key="+auth_key```

#this is the api call for the script to use to do the device group commit
```commit_devicegroup_api = 'https://panoramaname/api/?type=commit&action=all&cmd=<commit-all><shared-policy><device-group><entry name="YOURDG"/></device-group></shared-policy></commit-all>&key='+auth_key```

**Comment by jcrubaugh** [comment]
i assume you have tried something like this 

```panorama.commit(dg)```
